lua/bufferline/render.lua

```diff
@@ -3,34 +3,22 @@
 -- render.lua
 --
 
-local vim = vim
-local api = vim.api
-local nvim = require'bufferline.nvim'
-local utils = require'bufferline.utils'
-local icons = require'bufferline.icons'
-local state = require'bufferline.state'
 local Buffer = require'bufferline.buffer'
-local Layout = require'bufferline.layout'
+local icons = require'bufferline.icons'
 local JumpMode = require'bufferline.jump_mode'
-local get_icon = icons.get_icon
-local len = utils.len
-local slice = utils.slice
-local strwidth = nvim.strwidth
-local reverse = utils.reverse
-local has = vim.fn.has
-local bufnr = vim.fn.bufnr
-local strcharpart = vim.fn.strcharpart
-local getbufvar = vim.fn.getbufvar
-local tabpagenr = vim.fn.tabpagenr
-
+local Layout = require'bufferline.layout'
+local state = require'bufferline.state'
+local strwidth = vim.api.nvim_strwidth
+local substring = string.sub
+local utils = require'bufferline.utils'
 
 local HL_BY_ACTIVITY = {
   [0] = 'Inactive',
   [1] = 'Visible',
   [2] = 'Current',
 }
 
-local function hl(name)
+local function tabline_hl(name)
```

Comment:
`hl` is commonly used as a variable to describe `group[1]`. I renamed the function to avoid shadowing the variables and vice versa.